hbase-server/src/test/java/org/apache/hadoop/hbase/client/TestAsyncRegionAdminApi.java

```diff
@@ -529,4 +529,28 @@ static void loadData(final TableName tableName, final byte[][] families, final i
       puts.clear();
     }
   }
+
+  @Test
+  public void testCompactionState() throws IOException, InterruptedException {
+    TableName tableName = TableName.valueOf("testCompactionState");
+    byte[] family = Bytes.toBytes("CF");
+    TableDescriptor td = TableDescriptorBuilder.newBuilder(tableName)
+      .setColumnFamily(ColumnFamilyDescriptorBuilder.of(family)).build();
+    final Admin admin = TEST_UTIL.getAdmin();
+    if (admin.tableExists(tableName)) {
+      admin.disableTable(tableName);
+      admin.deleteTable(tableName);
+    }
+    admin.createTable(td, new byte[][] { Bytes.toBytes(1) });
+    TEST_UTIL.waitTableAvailable(tableName);
+    TEST_UTIL.waitUntilNoRegionsInTransition(10000);
+    byte[] regionName = admin.getRegions(tableName).get(0).getRegionName();
+    admin.unassign(regionName, true);
+    TEST_UTIL.waitUntilNoRegionsInTransition(10000);
+    try {
```

Comment:
```
    try {
      admin.getCompactionState(tableName);
    } catch (IOException e) {
      fail("Exception is not excepted.");
    }
```
is the same as
```
admin.getCompactionState(tableName);
```
Tests fails in case of exceptions.